### PR TITLE
update export.md: clarify that constructor proxies are exported

### PR DESCRIPTION
**Pull request by unkarjedy**
Currently, the docs say:
> A member is eligible if all of the following holds:
> ...
> it is not a constructor, nor the (synthetic) class part of an object,

However, constructor proxies are exported and they are synthetic

related: https://github.com/lampepfl/dotty/issues/11128

### CODE REVIEW

File: docs/docs/reference/other-new-features/export.md [hunk 1]
```diff
@@ -69,6 +69,7 @@ A member is _eligible_ if all of the following holds:
    a base class of the class containing the export clause.
  - it is accessible at the export clause,
  - it is not a constructor, nor the (synthetic) class part of an object,
+ - it is a constructor proxy
```
Comment (odersky) [inline docs/docs/reference/other-new-features/export.md:72]: That implies that *only* constructor proxies can be exported.

**Review by odersky — CHANGES_REQUESTED** [review]

**Comment by odersky** [comment]
> However, constructor proxies are exported and they are synthetic

But constructor proxies are not the synthetic class part of an object. So the original  wording is correct.

File: docs/docs/reference/other-new-features/export.md (hunk 1 above)
Comment (unkarjedy) [reply docs/docs/reference/other-new-features/export.md:72]: Indeed it should have been appended to previous item as a note

**Comment by unkarjedy** [comment]
@odersky 
It still feels that constructor proxies could be mention somewhere more explicitly.

>But constructor proxies are not the synthetic class part of an object. So the original wording is correct.

There is still this part in the docs:
>A wildcard selector * creates aliases for all eligible members of path **except** for given instances, **synthetic members generated by the compiler** and those members that are named by a previous simple, renaming, or omitting selector.

Aren't constructor proxies also "synthetic members generated by the compiler"?

**Comment by unkarjedy** [comment]
Also regarding this part:
>A wildcard selector * creates aliases for all eligible members of path except for given instances, synthetic members generated by the compiler and those members that are named by a previous simple, renaming, or omitting selector.
A member is eligible if all of the following holds:

Aren't the members, generated by the compiler for `export` clauses considered synthetic as well?

Currently, it's allowed to export already-exported members:
```scala
object Outer1:
  object Inner:
    def foo = 1

  export Inner._

  println(foo)

object Outer2:
  export Outer1._

  println(foo)
```

Maybe this case could also be clarified in the docs?

**Comment by odersky** [comment]
Yes, indeed. That sentence needs changing. Maybe:

> A wildcard selector * creates aliases for all eligible members of path except for given instances, synthetic members generated by the compiler and those members that are named by a previous simple, renaming, or omitting selector. 

> Notes: 
>  - eligible construtor proxies are also included, even though they are synthetic members. 
>  - members created by an export are also included. They are created by the compiler, but are not considered synthetic.

**Comment by unkarjedy** [comment]
Great!
I will patch the working in the PR

**Comment by unkarjedy** [comment]
done

**Review by bishabosha — APPROVED** [review]
LGTM, thanks!

**Pull request merged**